pandas/src/ujson/python/objToJSON.c

```diff
@@ -2065,6 +2065,23 @@ void Object_beginTypeContext (JSOBJ _obj, JSONTypeContext *tc)
     pc->PyTypeToJSON = NpyFloatToDOUBLE; tc->type = JT_DOUBLE;
     return;
   }
+  else
+  if (PyArray_Check(obj) && PyArray_CheckScalar(obj)) {
+    #if PY_MAJOR_VERSION >= 3
+      PyErr_Format(
+        PyExc_TypeError,
```

Comment:
well your comment only applies to s 0-dim object array
anyother dtype could be handled as a scalar